src/inc/corinfo.h

```diff
@@ -1955,6 +1954,14 @@ typedef SIZE_T GSCookie;
 
 const int MAX_EnC_HANDLER_NESTING_LEVEL = 6;
 
+// Results from type comparison queries
+enum class TypeCompareState
+{
+    MustNot = -1, // types are not equal
+    May = 0,      // types may be equal (must test at runtime)
+    Must = 1,     // type are equal
```

Comment:
Another possibility here is:

Incongruent,
Unknown,
Congruent,

Reply:
In particular the usage of `May `is confusing as it represents the lack of knowledge.

Reply:
I forgot to mention this when we were chatting offline: these names match the ones used a in similar .NetNative extension of the jit interface. So am going to keep them as is...